### ISSUE TITLE
config section in ~/.cabal/config for the init workflow crashes when specifying newer cabal versions 

### ISSUE THREAD

**Issue by cartazio** [haskell/cabal#7300]
config section in ~/.cabal/config for the init workflow crashes when specifying newer cabal versions 

this is fine
```
init
  interactive: True
  -- cabal-version: 2.4
  -- license:
  -- tests:
  -- test-dir:
  -- language: Haskell2010
  -- application-dir: app
  -- source-dir: src
```

this is not

```
init
  interactive: True
  cabal-version: 3.2
  -- license:
  -- tests:
  -- test-dir:
  language: Haskell2010
  application-dir: app
  source-dir: src
```
yields this error
```
cabal: Error parsing config file /Users/wrn6115/.cabal/config:141:
Parse of field 'cabal-version' failed (Cannot parse Cabal specification
version: 3.2): 3.2
```

**Comment by fgaz** [comment]
cabal-version 3.2 [doesn't exist](https://cabal.readthedocs.io/en/3.4/file-format-changelog.html). the error could be nicer though (`Cannot parse Cabal specification
version: 3.2, this version doesn't exist, check this url...`).

Did cabal generate that section? That would be a bug

**Comment by cartazio** [comment]
Nope. I wanted to specify a version in the 3.x series and made one up.

What version values does this config suport?

**Comment by fgaz** [comment]
https://cabal.readthedocs.io/en/latest/file-format-changelog.html

**Comment by emilypi** [comment]
Just to reiterate my suggestion here: 

`cabal-version` describes a version *spec*, which makes the flag slightly misleading. I would propose two possible solutions: 
 
1. we look at calling this `cabal-spec` in the future, so as not to avoid confusion. This is a bit of a deep change, I"m not sure if i want to go down this route. 
2. we change the parser to locate a least upper spec bound for valid cabal versions.  No code really has to change besides the additional parsing cases. 

It is a rough edge as is. If it's tripping up power users like Carter, it's tripping up everyone else too.

**Comment by cartazio** [comment]
Even simply an enumeration of allowed spec versions and url for further
reading would be helpful

**Comment by fgaz** [comment]
Yeah, I think this is one of those cases where just a better error message goes a long way. @emilypi's second option is clever though

**Comment by cartazio** [comment]
In the latter case, a warning and the change suggestion/ reading ref would
still be nice.

**Comment by emilypi** [comment]
Yeah, a warning blurb e.g. like 

```
Warning: `cabal-version 3.2` defaults to the 3.0 parser specification. To remove these warnings, please set `cabal-version: 3.0` in your `.cabal` file.
```

Or something. I'm not a writer :)
